packages/core/core/src/services/document-service/internationalization.ts

```diff
@@ -37,26 +37,15 @@ const localeToLookup: Transform = (contentType, params) => {
   }
 
   if (params.locale) {
-    return assoc(['lookup', 'locale'], params.locale, params);
-  }
-
-  return params;
-};
+    if (typeof params.locale === 'string') {
+      if (params.locale === '*') {
+        return params;
+      }
 
-/**
- * Add locale lookup query to the params
- */
-const multiLocaleToLookup: Transform = (contentType, params) => {
-  if (!strapi.plugin('i18n').service('content-types').isLocalizedContentType(contentType)) {
-    return params;
-  }
-
-  if (params.locale) {
-    if (params.locale === '*') {
-      return params;
+      return assoc(['lookup', 'locale'], params.locale, params);
     }
 
-    return assoc(['lookup', 'locale'], params.locale, params);
+    return assoc(['lookup', 'locale', '$in'], params.locale, params);
```

Comment:
how does this work with relations?
I recall we used the document locale sometimes to transform relations, and I wonder if this might add some conflicts

Reply:
Do you mean specifically with a bulk locale publish (e.g. publish something that has a relation with another non i18n CT)?

Or do you mean for the other document service methods?
	-In which case, would making the change to only allows arrays in the params of (un)publish remove this risk?

Reply:
other documetn service methods, with the current typing, there might be users doing something like
`strapi.documents(uid).findOne(docId, { locale: ['en', 'fr']})`

It would not be breaking to allow that, but I don't think such api makes sense atm, so I would say for now I would only allow it on the (un)publish & discardDraft methods

Reply:
it can be as simple as throwing an error if passing something that is not a string on the other methods